### DISCUSSION TITLE
using GPT-4's pre-trained embeddings?

### DISCUSSION THREAD

**Discussion by chonchonj23** [MaartenGr/BERTopic#1435] [Q&A]
Dear Maarten, 

The code below has generated some excellent results, and I want to confirm if the code uses GPT-4's pre-trained embeddings.

from bertopic.representation import OpenAI
representation_model = OpenAI(model="gpt-4", chat=False)
model = BERTopic(embedding_model=representation_model) 
topics, probs = model.fit_transform(docs)

Best regards,
Chon

**Comment by MaartenGr** [comment]
It does not. What you are using is a representation model, not an embedding model. I would advise reading the [documentation](https://maartengr.github.io/BERTopic/getting_started/embeddings/embeddings.html#openai) instead.

**Reply by chonchonj23** [reply]
Thanks Maarten!

I can now see the magic a representation model can do on topic keywords!
